src/components/search/search.style.ts

```diff
@@ -148,38 +164,27 @@ const StyledSearch = styled.div<StyledSearchProps>`
         flex: 1;
         z-index: ${theme.zIndex.smallOverlay};
       }
+
       ${StyledIcon} {
-        ${darkVariant &&
-        css`
-          ${iconColor &&
+        :not([data-element="search"]) {
+          ${darkVariant &&
           css`
-            color: var(--colorsUtilityMajor400);
+            color: var(--colorsUtilityYang080);
```

Comment:
**a sudden realisation:** It might actually be this causing the issue that I pointed out in a previous comment. 
![Screenshot 2024-02-21 at 14 24 21](https://github.com/Sage/carbon/assets/56251247/240d51f6-e212-4ebc-88ed-788bdcb8a7a9)

Reply:
yeah I haven't handled for the searchButton in dark background basically should just be an additional check to ensure no styling changes when `searchButton` is truthy